### PR TITLE
dispose the watched wild card directories only if present

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes the crash and other side effects when config file without wildcards is removed from the disk

### CODE REVIEW

File: src/server/project.ts [hunk 1]
```diff
@@ -1059,10 +1059,12 @@ namespace ts.server {
                 this.typeRootsWatchers = undefined;
             }
 
-            this.directoriesWatchedForWildcards.forEach(watcher => {
-                watcher.close();
-            });
-            this.directoriesWatchedForWildcards = undefined;
+            if (this.directoriesWatchedForWildcards) {
```
Comment (andy-ms) [inline src/server/project.ts]: This property should have `| undefined` in its declaration then?

File: src/harness/unittests/tsserverProjectSystem.ts [hunk 2]
```diff
@@ -2335,6 +2335,32 @@ namespace ts.projectSystem {
...
+                typeAcquisition: { "include": [] }
+            }]);
+
+            let knownProjects = projectService.synchronizeProjectList([]);
+            host.reloadFS([libFile, site]);
+            host.triggerFileWatcherCallback(configFile.path);
+
+            knownProjects = projectService.synchronizeProjectList(map(knownProjects, proj => proj.info));
```
Comment (billti) [inline src/harness/unittests/tsserverProjectSystem.ts]: Worth verifying that the .js file got added back to the external project once the config file was deleted and it was removed from the configured project?

File: src/harness/unittests/tsserverProjectSystem.ts [hunk 3]
```diff
@@ -2335,6 +2335,32 @@ namespace ts.projectSystem {
...
+                rootFiles: [toExternalFile(configFile.path), toExternalFile(site.path)],
+                options: { "allowJs": false, "allowNonTsExtensions": false, "allowSyntheticDefaultImports": false, "allowUnreachableCode": false, "allowUnusedLabels": false, "alwaysStrict": false, "compileOnSave": true, "declaration": false, "emitBOM": false, "emitDecoratorMetadata": false, "experimentalAsyncFunctions": false, "experimentalDecorators": false, "forceConsistentCasingInFileNames": false, "importHelpers": false, "inlineSourceMap": false, "inlineSources": false, "isolatedModules": false, "jsx": 0, "noEmit": false, "noEmitHelpers": false, "noEmitOnError": true, "noFallthroughCasesInSwitch": false, "noImplicitAny": false, "noImplicitReturns": false, "noImplicitThis": false, "noImplicitUseStrict": false, "noLib": false, "noResolve": false, "noUnusedLocals": false, "noUnusedParameters": false, "preserveConstEnums": false, "removeComments": false, "skipDefaultLibCheck": false, "skipLibCheck": false, "sourceMap": true, "strictNullChecks": false, "stripInternal": false, "suppressExcessPropertyErrors": false, "suppressImplicitAnyIndexErrors": false, "target": 1 },
```
Comment (billti) [inline src/harness/unittests/tsserverProjectSystem.ts]: Might as well set allowJs to true, seeing as you have no TypeScript files and only a .js file, that way at least you'll have one source file in the configured project created/removed.

Comment (sheetalkamat) [inline src/harness/unittests/tsserverProjectSystem.ts]: Scenario requires it to be no files being picked by config file (as that's the only time wildcard directories are not watched)

**Pull request merged**